dokka-integration-tests/gradle/src/main/kotlin/org/jetbrains/dokka/it/gradle/AbstractGradleIntegrationTest.kt

```diff
@@ -120,12 +123,39 @@ abstract class AbstractGradleIntegrationTest : AbstractIntegrationTest() {
         val templateSettingsGradleKts: Path by systemProperty(Paths::get)
 
         /** file-based Maven repositories with Dokka dependencies */
-        private val devMavenRepositories: String by systemProperty { repos ->
-            val repoPaths = repos.split(",").map { Paths.get(it) }
+        private val devMavenRepositories: List<Path> by systemProperty { repos ->
+            repos.split(",").map { Paths.get(it) }
+        }
 
-            val reposSpecs = repoPaths
-                .withIndex()
-                .joinToString(",\n") { (i, repoPath) ->
+        private val mavenRepositories: String by lazy {
+            val reposSpecs = if (dokkaVersionOverride != null) {
+                // if `DOKKA_VERSION_OVERRIDE` environment variable is provided,
+                //  we allow running tests on a custom Dokka version from specific repositories
+                when {
+                    // release version like `2.0.0`
+                    !dokkaVersion.contains("-") -> "mavenCentral()"
+                    // locally published version for testing some bug like `2.0.0-local-reproducing-bug`
+                    dokkaVersion.contains("-local-") -> "mavenLocal()"
+                    // dev version like `2.0.0-dev-329`
+                    dokkaVersion.contains("-dev-") -> "maven(\"https://maven.pkg.jetbrains.space/kotlin/p/dokka/dev\")"
+                    // test version like `2.0.0-test-49477c44dfc58e2702f4c145ff41190b39d117fb`
+                    dokkaVersion.contains("-test-") -> "maven(\"https://maven.pkg.jetbrains.space/kotlin/p/dokka/test\")"
+                    else -> error(
+                        """
+                        Provided Dokka version override is not supported: $dokkaVersion.
+                        Supported versions are:
+                        - release versions like '2.0.0'
+                        - dev versions like `2.0.0-dev-329`
+                        - test versions like `2.0.0-test-49477c44dfc58e2702f4c145ff41190b39d117fb`
+                        - locally published (to mavenLocal) versions with `-local-` suffix like `2.0.0-local-reproducing-bug`
+                        """.trimIndent()
+                    )
+                }.also { repository ->
+                    println("Dokka version overridden with ${dokkaVersionOverride}. Using $repository for resolving Dokka")
+                }
```

Comment:
What about simplifying the logic here?

So long as the correct version of Dokka gets picked up, does it matter which repo are defined? The tests won't be as strict, but that's what the integration tests are for.

```suggestion
                println("Dokka version overridden with ${dokkaVersionOverride}")
                // if `DOKKA_VERSION_OVERRIDE` environment variable is provided,
                //  we allow running tests on a custom Dokka version from specific repositories
                """
                maven("https://maven.pkg.jetbrains.space/kotlin/p/dokka/test")
                maven("https://maven.pkg.jetbrains.space/kotlin/p/dokka/dev")
                mavenCentral()
                mavenLocal()
                """.trimIndent()
```

Reply:
Yeah, let's do it like this. Tried to be too careful :)